src/stan/services/variational/print_progress.hpp

```diff
@@ -0,0 +1,41 @@
+#ifndef STAN_SERVICES_VARIATIONAL_PRINT_PROGRESS_HPP
+#define STAN_SERVICES_VARIATIONAL_PRINT_PROGRESS_HPP
+
+#include <stan/services/io/do_print.hpp>
+#include <cmath>
+#include <iomanip>
+#include <iostream>
+#include <string>
+
+namespace stan {
+  namespace services {
+    namespace variational {
+
+      void print_progress(const int m,
+                          const int start,
+                          const int finish,
+                          const int refresh,
+                          const bool tune,
+                          const std::string prefix,
+                          const std::string suffix,
+                          std::ostream& o) {
+        int it_print_width = std::ceil(std::log10(static_cast<double>(finish)));
```

Comment:
Just to clutter up the issues more, thanks for taking care of all
this.  It makes a big difference to those of us in the maintenance
department!

> On Aug 26, 2015, at 11:51 AM, Alp Kucukelbir <notifications@github.com> wrote:
> 
> In src/stan/services/variational/print_progress.hpp:
> 
> > +#include <iostream>
> > +#include <string>
> > +
> > +namespace stan {
> > +  namespace services {
> > +    namespace variational {
> > +
> > +      void print_progress(const int m,
> > +                          const int start,
> > +                          const int finish,
> > +                          const int refresh,
> > +                          const bool tune,
> > +                          const std::string prefix,
> > +                          const std::string suffix,
> > +                          std::ostream& o) {
> > +        int it_print_width = std::ceil(std::log10(static_cast<double>(finish)));
> 
> i added error checking. this function will likely go away with the writer refactor anyhow.
> 
> —
> Reply to this email directly or view it on GitHub.
>